### ISSUE TITLE
issue testing across hpc nodes

### ISSUE THREAD

**Issue by scottma** [hpc/ior#471]
In the attached screenshot, I am running across 2 nodes (but tried running 1, 2 and 6 nodes). The result is the same as running 1 node regardless how many nodes I run the test on. Also, the Result (individual) is same as the overall test results. (with 1 iteration). Seems like with mpirun, it is only running in 1 node. feel like I am missing something here or it is just showing the stats of 1 node. anyone came across this?

command
#SBATCH --job-name=ior-perf-test
#SBATCH --output=%N_%x_%j_%t.out
#SBATCH --ntasks=32
#SBATCH --ntasks-per-node=16
export I_MPI_PIN_RESPECT_CPUSET=0
export I_MPI_HYDRA_BOOTSTRAP=slurm
module load intelmpi
mpirun bash "/s3mountpoint-speedtest/s3-mp-ior-run-separate.sh"

the invoking shellscript:

RUNNING_HOST=\`hostname\`
RUNNING_FILE=/\$RUNNING_HOST"/S@S@S"
/shared/ior/bin/ior -r -w -v -F -o=\$RUNNING_FILE -b=2560m -a=POSIX -i=1 -t=64m  

<img width="1439" alt="image" src="https://github.com/hpc/ior/assets/1713417/457fb328-4323-4b9d-b00c-6d1a423a1d14">

**Comment by JulianKunkel** [comment]
Can you clarify what you mean by the results were the "same"?

Looking at the output, it appears that nodes : 2 is correct here -> sometimes people use the wrong mpiexec/mpirun which causes MPI to run N independent processes. 

Is the output file maybe on a NFS server or sth. where performance doesn't increase with the processes?
As you run with -F the shared file access isn't the problem here.

**Comment by scottma** [comment]
Hi @JulianKunkel 

I actually figured this out. turned out not to be an IOR issue. What I was testing is the s3 mountpoint using IOR. There is a limit on throughput with s3 mountpoint, likely to be a bug in the underlying driver. So, when I tested it w/ IOR, regardless of the # of nodes performing the test,  I am getting the same throughput. (which is strange). At first, I thought it was a IOR reporting issue, but I tested it with EBS (e.g. local harddrive), it scaled correctly with IOR performance multiple by the number of nodes.

I submitted a different ticket for the s3 mountpoint team. We can close this ticket.
